nemo/collections/asr/data/audio_to_label.py

```diff
@@ -825,3 +997,761 @@ def sliced_seq_collate_fn(self, batch):
 
     def vad_frame_seq_collate_fn(self, batch):
         return _vad_frame_seq_collate_fn(self, batch)
+
+
+class _AudioDiarTrainDataset(Dataset):
+    """
+    Dataset class that loads a json file containing paths to audio files,
+    RTTM files and number of speakers. This Dataset class is designed for
+    training or fine-tuning speaker embedding extractor and diarization decoder
+    at the same time.
+
+    Example:
+    {"audio_filepath": "/path/to/audio_wav_0.wav", "num_speakers": 2,
+    "rttm_filepath": "/path/to/diar_label_0.rttm}
+    ...
+    {"audio_filepath": "/path/to/audio_wav_n.wav", "num_speakers": 2,
+    "rttm_filepath": "/path/to/diar_label_n.rttm}
+
+    Args:
+        manifest_filepath (str):
+            Path to input manifest json files.
+        clus_label_dict (Dict):
+            Segment-level speaker labels from Clustering results.
+        soft_label_thres (float):
+            Threshold that determines the label of each segment based on RTTM file information.
+        featurizer:
+            Featurizer instance for generating features from the raw waveform.
+        window_stride (float):
+            Window stride for acoustic feature. This value is used for calculating the numbers of feature-level frames.
+        emb_batch_size (int):
+            Number of embedding vectors that are trained with attached computational graphs.
+        max_spks (int):
+            Integer value that limits the number of speakers for the model that is being trained.
+        bi_ch_infer (bool):
+            This variable should be True if dataloader is created for an inference task.
+    """
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+        """
+        output_types = {
+            "features": NeuralType(('B', 'T'), AudioSignal()),
+            "feature_length": NeuralType(('B'), LengthsType()),
+            "ms_seg_timestamps": NeuralType(('B', 'C', 'T', 'D'), LengthsType()),
+            "ms_seg_counts": NeuralType(('B', 'C'), LengthsType()),
+            "clus_label_index": NeuralType(('B', 'T'), LengthsType()),
+            "scale_mapping": NeuralType(('B', 'C', 'T'), LengthsType()),
+            "targets": NeuralType(('B', 'T', 'C'), ProbsType()),
+        }
+
+        return output_types
+
+    def __init__(
+        self,
+        *,
+        manifest_filepath: str,
+        multiscale_args_dict: str,
+        ms_ts_dict: Dict,
+        soft_label_thres: float,
+        featurizer,
+        window_stride,
+        emb_batch_size,
+        max_spks: int,
```

Comment:
Isn't max_spkrs 2? , I don't think this should argument required, since based on collections.py, you prepare your data loader for 2 speakers always (line 679:683), so this max_spkrs doesn't mean anything

Reply:
sounds good